File: C programs/1448. Count Good Nodes in Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int DFS(struct TreeNode* node,int max)
{
    if(!node)
    {
        return 0;
    }
    int good=0;
    if(max <= node->val)
    {
        good++;
        max=node->val;
    }

    good+=DFS(node->left,max);
    good+=DFS(node->right,max);


    return good;
}


int goodNodes(struct TreeNode* root){

return DFS(root,root->val);
}

File: Neetcode 150/1448. Count Good Nodes in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(root,max):
            if not root:
                return 0
            cnt=0    
            if max<=root.val:
                cnt=1
                max=root.val
            cnt+=dfs(root.left,max)
            cnt+=dfs(root.right,max)
            return cnt    


        return dfs(root,root.val)    
        
